Makefile

```diff
@@ -52,6 +52,13 @@ check-services: $(CHECK_SERVICES_MODULES)
 pylint-%:
 	$(PYTHON) -m pylint --rcfile pylintrc --recursive=y $* --score=n
 
+.PRECIOUS: $(patsubst %, .venvs/%/bin/activate, $(SERVICES))
+.venvs/%/bin/activate: %/pinned-requirements.txt hail/python/dev/pinned-requirements.txt Makefile
+	$(PYTHON) -m venv .venvs/$*
+	source .venvs/$*/bin/activate && \
+	    pip3 install -r $*/pinned-requirements.txt -r hail/python/dev/pinned-requirements.txt && \
+	    pip3 install --editable ./gear ./web_common
```

Comment:
How come this doesn't need to install `hailtop` as editable?

Reply:
Dunno why this didn't cause problems, but I added it.